src/libxively/cbor/xi_cbor_codec_ct.c

```diff
@@ -121,55 +115,106 @@ void xi_cbor_codec_ct_encode( const xi_control_message_t* control_message,
             }
             break;
 
-        case XI_CONTROL_MESSAGE_CS_FILE_GET_CHUNK:
+        case XI_CONTROL_MESSAGE_CS__SFT_FILE_GET_CHUNK:
 
             xi_cbor_put_name_and_revision( cb_map, control_message->file_get_chunk.name,
                                            control_message->file_get_chunk.revision,
                                            &err );
 
             cn_cbor_map_put(
-                cb_map, cn_cbor_string_create( "O" CBOR_CONTEXT_PARAM, &err ),
+                cb_map,
+                cn_cbor_string_create(
+                    XI_CBOR_CODEC_CT_STRING_FILECHUNK_OFFSET CBOR_CONTEXT_PARAM, &err ),
                 cn_cbor_int_create(
                     control_message->file_get_chunk.offset CBOR_CONTEXT_PARAM, &err ),
                 &err );
 
             cn_cbor_map_put(
-                cb_map, cn_cbor_string_create( "L" CBOR_CONTEXT_PARAM, &err ),
+                cb_map,
+                cn_cbor_string_create(
+                    XI_CBOR_CODEC_CT_STRING_FILECHUNK_LENGTH CBOR_CONTEXT_PARAM, &err ),
                 cn_cbor_int_create(
                     control_message->file_get_chunk.length CBOR_CONTEXT_PARAM, &err ),
                 &err );
 
             break;
 
-        case XI_CONTROL_MESSAGE_CS_FILE_STATUS:
+        case XI_CONTROL_MESSAGE_CS__SFT_FILE_STATUS:
 
-            // todo_atigyi
+            xi_cbor_put_name_and_revision( cb_map, control_message->file_status.name,
+                                           control_message->file_status.revision, &err );
+
+            cn_cbor_map_put(
+                cb_map,
+                cn_cbor_string_create(
+                    XI_CBOR_CODEC_CT_STRING_FILESTATUS_PHASE CBOR_CONTEXT_PARAM, &err ),
+                cn_cbor_int_create( control_message->file_status.phase CBOR_CONTEXT_PARAM,
+                                    &err ),
+                &err );
+
+            cn_cbor_map_put(
+                cb_map,
+                cn_cbor_string_create(
+                    XI_CBOR_CODEC_CT_STRING_FILESTATUS_CODE CBOR_CONTEXT_PARAM, &err ),
+                cn_cbor_int_create( control_message->file_status.code CBOR_CONTEXT_PARAM,
+                                    &err ),
+                &err );
 
             break;
 
         /* the followings are encoded by the broker and decoded by the client */
-        case XI_CONTROL_MESSAGE_SC_FILE_UPDATE_AVAILABLE:
-        case XI_CONTROL_MESSAGE_SC_FILE_CHUNK:
-        default:
-            cn_cbor_free( cb_map CBOR_CONTEXT_PARAM );
-            return;
+        case XI_CONTROL_MESSAGE_SC__SFT_FILE_UPDATE_AVAILABLE:
+        case XI_CONTROL_MESSAGE_SC__SFT_FILE_CHUNK:
+        default:;
     }
 
-    unsigned char encoded[512];
-    *out_len = cn_cbor_encoder_write( encoded, 0, sizeof( encoded ), cb_map );
+    xi_state_t state = XI_STATE_OK;
+
+    uint8_t* encoded     = NULL;
+    uint32_t encoded_len = 256;
+
+    for ( ; encoded_len < 100000; encoded_len *= 2 )
```

Comment:
I would also allow for the starting value to be a preprocessor defintion, too, so that people can set them to be the same if they don't want to play too much with the heap.